src/fsharp/SyntaxTrivia.fsi

```diff
@@ -4,6 +4,34 @@ namespace FSharp.Compiler.SyntaxTrivia
 
 open FSharp.Compiler.Text
 
+[<RequireQualifiedAccess; NoEquality; NoComparison>]
+type ConditionalDirectiveTrivia =
+    | IfDirectiveTrivia of expr:IfDirectiveExpression * range:range
+    | ElseDirectiveTrivia of range:range
+    | EndIfDirectiveTrivia of range:range
+
+and [<RequireQualifiedAccess; NoEquality; NoComparison>] IfDirectiveExpression =
```

Comment:
In signature files just use "namespace rec" at the top of the file, then you don't need to use `and`.  There is never any real issue with doing this since there is no real inference or execution initialization going on in signature files.